src/tree/map.rs

```diff
@@ -885,7 +884,7 @@ macro_rules! define_iterator {
         item!(impl<'a, K, V> $name<'a, K, V> {
             #[inline(always)]
             fn next_(&mut self, forward: bool) -> Option<(&'a K, &'a $($addr_mut)* V)> {
-                while !self.stack.is_empty() || !self.node.is_null() {
```

Comment:
No, though I would expect it to be faster since each condition is only checked once. Let me get some metrics.